cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Given a column-view of strings type, an instance of this class
+ * provides the strings operations on the column.
+ *---------------------------------------------------------------------------**/
+class strings_column_handler
```

Comment:
Can we just call this a `strings_column_view`? I think that would be more consistent with the overall naming. It's a view into a column of strings and defines strings specific stuff. 

Furthermore, I think it should inherit from `column_view` so that you get all of the APIs that `column_view` already provides. For instance, this class is already missing some useful APIs that `column_view` provides, such as `offset()`, `has_nulls()`, etc. 

There's probably some APIs in `column_view` that you don't want to expose in `strings_column_view`, such as `data()`. Here's how we can do this:

```
class strings_column_view : private column_view { // Private inheritance from column_view

 
   // This will make available only the APIs you explicitly specify 
   using column_view::nullable;
   using column_view::size;
   using column_view::has_nulls;
   // etc.
};

Otherwise, if any of those APIs change, you'll have to manually update everything here.